From 16f229f08fe504b3f161d309cc68062938f8600b Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 30 Jun 2022 13:29:01 +0700
Subject: [PATCH 06/27] soc: renesas: Fix mistake that overwrite revision of
 wrong target

The product revision information is overwritten in case current SoC
is RZG2UL. However, the strcmp() function was used wrongly when trying
to detect RZG2UL. Due to this mistake, product revision of other SoCs
are overwritten instead.
strcmp returns 0 when the strings match, so its result must be reverted.

Signed-off-by: Hung Tran <hung.tran.jy@renesas.com>
Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 drivers/soc/renesas/renesas-soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/renesas/renesas-soc.c b/drivers/soc/renesas/renesas-soc.c
index 467c1d438d3d..4eac5ffe6a2e 100644
--- a/drivers/soc/renesas/renesas-soc.c
+++ b/drivers/soc/renesas/renesas-soc.c
@@ -448,7 +448,7 @@ static int __init renesas_soc_init(void)
 						   eslo);
 
 	/*FIXME: current RZG2UL family do not support product revision*/
-	if (strcmp(soc_dev_attr->family, fam_rzg2ul.name))
+	if (!strcmp(soc_dev_attr->family, fam_rzg2ul.name))
 		soc_dev_attr->revision = 0;
 
 	pr_info("Detected Renesas %s %s %s\n", soc_dev_attr->family,
-- 
2.17.1

